packages/fulfillment/src/services/fulfillment-module-service.ts

```diff
@@ -671,7 +748,117 @@ export default class FulfillmentModuleService<
   ): Promise<
     FulfillmentTypes.ShippingOptionDTO[] | FulfillmentTypes.ShippingOptionDTO
   > {
-    return []
+    const updatedShippingOptions = await this.updateShippingOptions_(
+      data,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<
+      FulfillmentTypes.ShippingOptionDTO | FulfillmentTypes.ShippingOptionDTO[]
+    >(updatedShippingOptions, {
+      populate: true,
+    })
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async updateShippingOptions_(
+    data:
+      | FulfillmentTypes.UpdateShippingOptionDTO[]
+      | FulfillmentTypes.UpdateShippingOptionDTO,
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<TShippingOptionEntity | TShippingOptionEntity[]> {
+    const dataArray = Array.isArray(data) ? data : [data]
+
+    if (!dataArray.length) {
+      return []
+    }
+
+    const shippingOptionIds = dataArray.map((s) => s.id)
+    if (!shippingOptionIds.length) {
+      return []
+    }
+
+    const shippingOptions = await this.shippingOptionService_.list(
+      {
+        id: shippingOptionIds,
+      },
+      {
+        relations: ["rules"],
+      },
+      sharedContext
+    )
+
+    FulfillmentModuleService.validateMissingShippingOptions_(
```

Comment:
Why is this a static method? For validation, we typically use regular methods. You've done that with `validateRules`, so I would consider changing this to be consistent

Reply:
validateRules is a utils that leaves with the "rule engine" it is not a method. Also, I use static as it is not using anything from the class context but is tight to the module data

Reply:
> validateRules is a utils that leaves with the "rule engine" it is not a method

Sorry, this was an oversight from my side. 

I am still not super fond of the static methods. Haven't we generally avoided those in the past?